lms/djangoapps/mobile_api/friends/views.py

```diff
@@ -0,0 +1,96 @@
+
+    **Example request**:
+
+        GET /api/mobile/v0.5/friends/course/<course_id>
+
+        where course_id is in the form of /edX/DemoX/Demo_Course
+
+    **Response Values**
+
+        {   "friends":
+                [{
+                    "name": "test",
+                    "id": "12345",
+                    "edX_id": "678910"
+                },
+                ...
+                ]
+            }
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.FriendsInCourseSerializer
+
+    def list(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.GET, files=request.FILES)
+        if serializer.is_valid():
+            # Get all the users FB friends
+            graph = facebook.GraphAPI(serializer.object['oauth_token'])
+            url = _FACEBOOK_API_VERSION + "me/friends"
+            friends = graph.request(url)
+            data = self.get_pagination(friends)
+            # For each friend check if they are a linked edX user
+            friends_that_are_edX_users = []
+            for friend in data:
+                query_set = UserSocialAuth.objects.filter(uid=unicode(friend['id']))
+                if query_set.count() == 1: 
+                    friend['edX_id'] = query_set[0].user_id
+                    friend['edX_username'] = query_set[0].user.username
+                    friends_that_are_edX_users.append(friend)
+            # Filter by sharing preferences
+            friends_that_are_edX_users_with_sharing = [friend for friend in friends_that_are_edX_users if self.sharing_pref_true(friend)]
+            if ('course_id' in kwargs) and len(kwargs['course_id'].split('/')) == 3:
+                course_path = kwargs['course_id'].split('/')
+                course_key = SlashSeparatedCourseKey(course_path[0], course_path[1], course_path[2])
+            fb_friends_in_course = [friend for friend in  friends_that_are_edX_users_with_sharing if self.is_member(course_key, friend)]
+            return Response({'friends' : fb_friends_in_course})
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+    def is_member(self, course_key, friend):
+            return CourseEnrollment.objects.filter(course_id = course_key, 
+                                                        user_id =  friend['edX_id'], 
+                                                        is_active = True ).count() == 1
+
+    def get_pagination(self, friends): 
+        '''
+            Get paginated data from response
+        '''
+        data = friends['data']
+        while 'paging' in friends and 'next' in friends['paging']: 
+            response = urllib2.urlopen(friends['paging']['next'])
+            friends = json.loads(response.read())
+            data = data + friends['data']
+        return data
+
+    def sharing_pref_true(self, friend):
```

Comment:
This method is also defined in the `courses` django app.  Can we reuse one implementation in both?  If the `courses` endpoint is moved into this app anyway, that'll solve it.